azafea/event_processors/endless/metrics/tests/test_events.py

```diff
@@ -431,6 +483,25 @@ def test_hack_clubhouse_progress_event_with_unknown_key():
     assert not hasattr(event, 'unknown')
 
 
+def test_invalid_parental_controls_changed_event():
+    from azafea.event_processors.endless.metrics.events import ParentalControlsChanged
+
+    # Make an invalid payload with missing keys
+    payload = GLib.Variant('mv', GLib.Variant('a{sv}', {
+        'AllowUserInstallation': GLib.Variant('b', False),
+        'AllowSystemInstallation': GLib.Variant('b', False),
+        'UnexpectedKey': GLib.Variant('s', 'should be ignored'),
+    }))
+
+    with pytest.raises(ValueError) as excinfo:
+        ParentalControlsChanged(payload)
+
+    assert str(excinfo.value) == ('Metric event 449ec188-cb7b-45d3-a0ed-291d943b9aa6 needs an '
```

Comment:
Personally I'm not a fan of making assertions about the exact human-readable text in assertion errors, but I see there are other instances of that already in this file!

Reply:
Yeah, this was copied from another test, so I’d prefer to not do anything radical in this commit to change it. There isn’t actually much else to go on when comparing `ValueError`s, so I guess the fix would be a slightly more invasive use of a different exception class.